auth/token-managers/jwt-token-manager.ts

```diff
@@ -70,33 +70,30 @@ export class JwtTokenManager {
   }
 
   /**
-   * This function sends an access token back through a callback. The source of the token
-   * is determined by the following logic:
+   * This function returns a Promise that resolves with an access token, if successful.
+   * The source of the token is determined by the following logic:
    * 1. If user provides their own managed access token, assume it is valid and send it
    * 2. a) If this class is managing tokens and does not yet have one, make a request for one
    *    b) If this class is managing tokens and the token has expired, request a new one
    * 3. If this class is managing tokens and has a valid token stored, send it
    *
-   * @param {Function} cb - callback function that the token will be passed to
    */
-  public getToken(cb: Function) {
+  public getToken(): Promise<any> {
     if (!this.tokenInfo[this.tokenName] || this.isTokenExpired()) {
```

Comment:
I think this should be wrap in a promise like my example above

Reply:
Since the `requestToken` method also returns a Promise, I think it is cleaner to chain the Promises together than to create a new Promise altogether.

Reply:
What happens if `this.tokenInfo` is undefined?

Reply:
It would throw an error that would not be rejected in the Promise, so the users rejection handling wouldn't catch it. Which would be bad. But we explicitly define the `tokenInfo` object in the constructor: `this.tokenInfo = {};`.

I understand what you're saying, I just don't know how much we want to account for a scenario like this that shouldn't ever happen. We were not accounting for any errors happening here before either. The behavior is staying the same. That said, if you think we need to be able to catch errors there, I will change the code